prettier, eslint 설정 파일 추가 및 pre-commit 훅 설정했습니다.

changheedev opened the pull request:
`.vscode/settings.json` 동작 확인하고 공통으로 적용할 prettier 설정은 루트 디렉토리에 생성했습니다.

그리고 commit 단계에서 prettier 포맷팅과 eslint fix 작업 처리하는 pre-commit 훅 설정했습니다.

prettier는 회의할 때 확인했던 설정 그대로 적용하였는데 eslint 설정은 거의 기본상태입니다. 

혹시 더 추가하면 좋을거 있으면 리뷰남겨주세요.

changheedev committed:
Chore: prettier, eslint 설정 파일 추가

코드 포맷을 통일하기 위한 lint 설정

yejineee commented on server/package.json:
```diff
@@ -8,13 +8,37 @@
+    "*.js": [
+      "prettier --write",
+      "eslint --fix",
+      "git add"
```
lint-staged는 처음 알게 되었습니다!
prettier와 eslint로 코드 교정까지 해주니까 좋을 것 같아요 👍

changheedev replied:
전체 포맷팅이 아니라 staging 된 부분만 포맷팅을 해주기 위해 사용했습니다. ㅎㅎ

아 그리구 지금 eslint 설정은 거의 기본 설정인데 추천해주실 rule 있으시면 남겨주세요.

rolled-potatoes replied:
저도 이거 처음봤는데 괜찮은거 같네요 .

그런데 gitignore 파일 수정할 때는 추가적으로 `--no-verify` 를 붙여줘야 하더라고요.

changheedev replied:
음.. gitignore 파일 포맷팅 말씀하시는건가요? 
js파일 외에는 *로 한번에 처리해버렸는데 말씀하신부분 한번 찾아보고 수정이 필요하면 수정해볼께요 👍

yejineee commented on server/.eslintrc.json:
```diff
@@ -0,0 +1,11 @@
+    "es6": true
+  },
+  "plugins": ["prettier"],
+  "extends": ["airbnb-base", "eslint:recommended", "prettier"],
```
`eslint-config-prettier`는 포맷팅 규칙을 prettier가 하게 하고, 중복되지 않은 규칙에 대해서 검사를 한다고 합니다.  `eslint-plugin-prettier`는 eslint규칙에 맡게 실행하고, 오류를 eslint의 오류로 나타나게 해준다고 합니다. 이 두 패키지를 한 번에 적용시키는 것이 'plugin:prettier/recommended'라고 합니다. 제가 참고한 블로그인데, [eslint-prettier-airbnb-style-guide로 설정하기](https://velog.io/@_jouz_ryul/ESLint-Prettier-Airbnb-Style-Guide%EB%A1%9C-%EC%84%A4%EC%A0%95%ED%95%98%EA%B8%B0) 여기 2.2, 2.3에 관련 내용이 있습니다. 확인해보시고, 한 번 고쳐보는건 어떨까요?

changheedev replied:
오 그렇군요! 

eslint 설정이 복잡하고 참고하는 자료마다도 달라서 헷갈렸는데 변경해서 테스트 해볼께요

yejineee commented on server/.eslintrc.json:
```diff
@@ -0,0 +1,11 @@
+{
+  "env": {
+    "node": true,
+    "es6": true
```
![스크린샷 2020-10-26 오후 10 11 29](https://user-images.githubusercontent.com/43772082/97176384-3422c200-17d8-11eb-9c45-9b75d959a332.png)
console 사용하셨을 때 에러 뜨셨던게, "browser": true를 설정 안하셔서 그런건가 싶네요..!
출처 : [esling-config](https://velog.io/@kyusung/eslint-config-2)

changheedev replied:
제가 겪었던 console 에러부분은 airbnb 쪽 lint `no-console` 설정이 `error` 로 설정되어 있는데 `console.log` 를 사용해서였습니다...😢 근데 이상하게 이번에는 airbnb 린트를 추가해도 warning 으로 처리되고 있네요. 이유를 찾아보아야 할 것 같습니다.

yejineee commented on server/.eslintrc.json:
```diff
@@ -0,0 +1,11 @@
+  "extends": ["airbnb-base", "eslint:recommended", "prettier"],
+  "rules": {
+    "prettier/prettier": ["error"]
+  }
```
{
    "no-use-before-define": ["error", { "functions": false}]
}
함수 선언하기 전에는 사용할 수 없다는 에러인데, 함수는 호이스팅이 일어나서, 에러 안뜨게 해줘도 된다고 합니다! 
[no-use-before-define](https://eslint.org/docs/rules/no-use-before-define)

changheedev replied:
추가했습니다. 감사합니다.

changheedev committed:
Refactor: eslint 설정 및 lint-staged 설정 수정

- eslint-config-prettier, eslint-plugin-prettier 모듈 설정시 plugins 설정과 extends 설정 대신 plugin:prettier/recommended 설정 하나로 처리하는 방식 적용

- 함수 호이스팅과 관련하여 no-use-before-define 옵션 설정 추가

- '.js' 파일 외의 lint 대상 파일을 '*' 대신 '*.md' 로 명시적으로 설정
